### ISSUE TITLE
Switch case errors 

### ISSUE THREAD

**Issue by cleptric** [squizlabs/PHP_CodeSniffer#1313]
PHP_CodeSniffer version 2.7.1 - PHP 7.1.0

```
switch ($i) {
    case 0:
        echo "i equals 0";
        break;
    case 1:
        echo "i equals 1";
        break;
    case 2:
        echo "i equals 2";
        break;
}  
```

gives me `Notice: Undefined index: scope_opener` and `PHP Notice:  Undefined index: scope_closer` errors. The result is:

```
2 | ERROR | CASE statements must be defined using a colon
4 | ERROR | CASE statements must be defined using a colon
```

**Comment by gmponos** [comment]
I think it would be more helpful if you could output the sniff name. I think `-s ` switch in your command.

**Comment by cleptric** [comment]
It's the `SwitchDeclarationSniff`

**Comment by jrfnl** [comment]
@cleptric I've tried to reproduce the issue, but failed. Tried on PHPCS 2.7.1 + master with PHP 5.6.28 and PHP 7.1.0.

Are you sure that is the code which is triggering the notice ? Is it a nested switch maybe ? Please provide a sample with which the issue can be consistently reproduced and a stack trace of the error notice would be nice too.

**Comment by gsherwood** [comment]
What @jrfnl said.

These problems are often caused by a file tokenizing problem due to nested statements, so I'd probably need the code around this statement to diagnose the problem. If possible, a link to the entire file would be perfect as I can cut things out and move things around until I find the smallest bit of code to reproduce the problem.

**Comment by cleptric** [comment]
@gsherwood 

```
<?php

namespace App;

class Foo
{

    /**
     * Genrate a formatted list
     *
     * @param array $data data
     * @param string|null $formattedList Formatted list
     * @return array|null
     */
    private function __outputFormattedList(?array $data, array $formattedList = null): ?array
    {
        foreach ($data as $key => $entry) {
            switch ($entry->type) {
                case Object::FOO:
                    break;
                case Object::BAR:
                    break;
                default:
                    throw new InvalidTypeException("{$entry->type} is not a valid object type!");
            }
        }

        return $formattedList;
    }
}
```

This is all I can send you

**Comment by jrfnl** [comment]
@cleptric Ok, that's useful.

I still cannot reproduce the undefined index notices, but I can reproduce the sniff incorrectly throwing the `CASE statements must be defined using a colon` error.

This was caused by the tokenizer getting confused over the nullable returntype which caused it to identify the `T_COLON` after the case statement as an `T_INLINE_ELSE`.

It looks like this has been fixed in `master` by the changes to the tokenizer which added the `T_NULLABLE` token type.

The most minimal example distilled from your code to reproduce the issue is:
```php
function __outputFormattedList($data, $formattedList): ?array
{
	switch ($data) {
		case Object::FOO:
			break;
	}
}
``` 

If the return type is changed to `array` (not nullable), the problem can no longer be reproduced.

**Comment by cleptric** [comment]
@jrfnl Let me try to get you a stripped out version where the undefined index notices occur :)
Thx so far!

**Comment by cleptric** [comment]
Ok, I just installed dev-master, and everything is fine now. No more `undefined_index` and other errors.
Waiting for the release :)

**Comment by jrfnl** [comment]
Issue turned out to be a duplicate of #1279
